### PR TITLE
Show subject viewer in feedback

### PR DESCRIPTION
**Pull request by rogerhutchings**
Staging branch URL: https://show-subject-viewer-in-feedback.pfe-preview.zooniverse.org/

Test project URL: https://feedback-update.pfe-preview.zooniverse.org/projects/rogerhutchings/feedback-test/classify

Following on from #4191, this adds a subject viewer into the feedback modal to show where target areas are for the radial reduction strategy, woo! It requires #4191 merging first.

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by rogerhutchings** [comment]
I should also fess up that this produces a whole bunch of console errors when the subject viewer is first mounted, but it does work - I think there's some missing default props somewhere in the tools.

File: app/classifier/tasks/drawing/markings-renderer.cjsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 React = require 'react'
 drawingTools = require '../../drawing-tools'
+`import strategies from '../../../features/feedback/shared/strategies';`
```
Comment (eatyourgreens) [inline app/classifier/tasks/drawing/markings-renderer.cjsx]: Backticked imports in coffeescript break the tests with `illegal import statement` errors. Use `require(…).default` instead.

**Comment by eatyourgreens** [comment]
Looks like this needs a big old rebase because of #4191

**Comment by rogerhutchings** [comment]
Done!

File: app/features/feedback/classifier/components/feedback-modal.jsx [hunk 2]
```diff
@@ -11,14 +13,14 @@ counterpart.registerTranslations('en', {
...
+function FeedbackModal({ messages, subjectViewerProps }) {
   return (
-    <section>
+    <section className="feedbackmodal">
       <Translate content="FeedbackModal.title" component="h2" />
+      {subjectViewerProps && (<SubjectViewer {...subjectViewerProps} />)}
       <ul>
         {messages.map(message =>
-          <li key={message}>
+          <li key={Math.random()}>
```
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply app/features/feedback/classifier/components/feedback-modal.jsx]: Elsewhere in PFE, the random key is generated once for a given list item, which is the correct behaviour. This code generates a new key on each render, which will prevent React from correctly tracking changes to the associated item.

[the remark this answers is not in the document]
Comment (rogerhutchings) [reply app/features/feedback/classifier/components/feedback-modal.jsx]: The list doesn't change tho

[the remark this answers is not in the document]
Comment (mcbouslog) [reply app/features/feedback/classifier/components/feedback-modal.jsx]: [The article linked](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318) notes the following exceptions, where using the array index in the key may be ok:
```
1. the list and items are static–they are not computed and do not change;
2. the items in the list have no ids;
3. the list is never reordered or filtered.
```
I think this list of feedback info qualifies as an exception? So `Math.random()` is ok, or the array index with the message (maybe shortened) would also work.

[the remark this answers is not in the document]
Comment (mcbouslog) [reply app/features/feedback/classifier/components/feedback-modal.jsx]: I'm thinking merge as-is and we can re-visit?

**Comment by mcbouslog** [comment]
This LGTM, I'm going to merge and we can revisit keys if need be with a new issue.

**Pull request merged**